asreview/webapp/utils/project.py

```diff
@@ -321,32 +322,54 @@ def get_statistics(project_id):
 
 
 def export_to_string(project_id, export_type="csv"):
-    fp_lock = get_lock_path(project_id)
+
+    # read the dataset into a ASReview data object
     as_data = read_data(project_id)
+
+    # set the lock to safely read labeled, pool, and proba
+    fp_lock = get_lock_path(project_id)
     with SQLiteLock(
-            fp_lock, blocking=True, lock_name="active", project_id=project_id):
+            fp_lock,
+            blocking=True,
+            lock_name="active",
+            project_id=project_id
+    ):
         proba = read_proba(project_id)
-        if proba is None:
-            proba = np.flip(np.arange(len(as_data)))
-        else:
-            proba = np.array(proba)
-        labels = read_current_labels(project_id, as_data=as_data)
-
-    pool_idx = np.where(labels == LABEL_NA)[0]
-    one_idx = np.where(labels == 1)[0]
-    zero_idx = np.where(labels == 0)[0]
+        pool = read_pool(project_id)
+        labeled = read_label_history(project_id)
 
-    proba_order = np.argsort(-proba[pool_idx])
-    ranking = np.concatenate((one_idx, pool_idx[proba_order], zero_idx),
-                             axis=None)
+    # get the record_id of the inclusions and exclusions
+    inclusion_record_id = [int(x[0]) for x in labeled if x[1] == 1]
+    exclusion_record_id = [int(x[0]) for x in labeled if x[1] == 0]
 
+    # order the pool from high to low proba
+    if proba is not None:
+        pool_ordered = proba.loc[pool, :] \
+            .sort_values("proba", ascending=False).index.values
+    else:
+        pool_ordered = pool_ordered
+
+    # get the ranking of the 3 subcategories
+    ranking = np.concatenate(
+        (
+            # add the inclusions first
+            inclusion_record_id,
+            # add the ordered pool second
```

Comment:
where can I check how the inclusion_record_id is ordered? should be in order seen (label_history?), or is it on the proba of the last model?

Reply:
same for exclusion_record_id

Reply:
Inclusions are ordered by order seen. Same for exclusions. Everything in between is ordered on proba (if available).